agent/agent.go

```diff
@@ -1047,13 +1047,6 @@ func (a *Agent) finishTask(ctx context.Context, tc *taskContext, status string,
 
 	a.killProcs(ctx, tc, false, "task is ending")
 
-	if tc.logger != nil {
-		tc.logger.Execution().Infof("Sending final task status: '%s'.", detail.Status)
-		flushCtx, cancel := context.WithTimeout(ctx, time.Minute)
-		defer cancel()
-		grip.Error(errors.Wrap(tc.logger.Flush(flushCtx), "flushing logs"))
-	}
-
 	grip.Infof("Sending final task status: '%s'.", detail.Status)
 	resp, err := a.comm.EndTask(ctx, detail, tc.task)
 	if err != nil {
```

Comment:
The return here might cause the logs not to be flushed which we want to avoid

Reply:
ah, I see we must flush twice as the ticket says